tests/test_enterprise/test_signals.py

```diff
@@ -625,3 +626,61 @@ def test_update_catalog_with_existing_service_catalog(self, api_client_mock):
             enabled_course_modes=enterprise_catalog.enabled_course_modes,
             publish_audit_enrollment_urls=enterprise_catalog.publish_audit_enrollment_urls
         )
+
+    @mock.patch('enterprise.signals.EnterpriseCatalogApiClient')
+    def test_update_enterprise_query(self, api_client_mock):
+        content_filter_1 = {
+            'content_type': 'course',
+        }
+
+        content_filter_2 = {
+            'content_type': 'course2',
+        }
+
+        sender = mock.Mock()  # this would be an EnterpriseCatalogQuery class
+
+        # kwargs to satisfy parameter
+        kwargs = {
+            'update_fields': None,
+            'raw': False,
+            'signal': '<django.db.models.signals.ModelSignal object at test>',
+            'using': 'default',
+            'created': False,
+        }
+
+        test_query = EnterpriseCatalogQueryFactory()  # represents instance of EnterpriseCatalogQuery
+        test_query.content_filter = content_filter_1
+        test_query.save()
+
+        enterprise_catalog_1 = EnterpriseCustomerCatalogFactory()
+        enterprise_catalog_2 = EnterpriseCustomerCatalogFactory()
+
+        enterprise_catalog_1.sync_enterprise_catalog_query = True
+        enterprise_catalog_2.sync_enterprise_catalog_query = True
+        enterprise_catalog_1.enterprise_catalog_query = test_query
+        enterprise_catalog_2.enterprise_catalog_query = test_query
+        enterprise_catalog_1.save()
+        enterprise_catalog_2.save()
+
+        test_query.content_filter = content_filter_2
+
+        update_enterprise_query(sender, test_query, **kwargs)  # calls post_save signal
+
+        self.assertEqual(enterprise_catalog_1.content_filter, enterprise_catalog_2.content_filter, content_filter_2)
+
+        enterprise_catalog_1.save()
+        enterprise_catalog_2.save()
+
+        api_client_mock.return_value.get_enterprise_catalog.return_value = True
+
+        # verify that the mock api was called when saving the catalog after updating the query
+        # enterprise_catalog_2 was most recently modified so the last call should be for that
+        api_client_mock.return_value.update_enterprise_catalog.assert_called_with(
```

Comment:
Just a note that this is a good thing to test thoroughly on stage, as we definitely want to be sure the sync still happens